test/groovy/PiperStageWrapperTest.groovy

```diff
@@ -320,4 +320,47 @@ class PiperStageWrapperTest extends BasePiperTest {
         assertThat(DebugReport.instance.failedBuild.fatal, is('true'))
         assertThat(DebugReport.instance.failedBuild.reason, is(caught))
     }
+
+    @Test
+    void testStashing() {
+
+        String stageNameStashStageFiles, stageNameUnstashStageFiles
+        List stashConfigStash, stashConfigUnstash
+
+        def utilsMock =  new com.sap.piper.Utils() {
+
+            public unstashStageFiles(Script script, String stageName,  List stashes = []) {
+                stageNameUnstashStageFiles = stageName
+                stashConfigUnstash = script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.unstash ?: []
+            }
+            public stashStageFiles(Script script, String stageName)  {
+                stageNameStashStageFiles = stageName
+                stashConfigStash = script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.stash ?: []
+            }
+            public void pushToSWA(Map parameters, Map config) {
+
+            }
+        }
+
+        nullScript.commonPipelineEnvironment.configuration['stageStashes'] = [
+            foo: [
+                stash: ['foo-stash'],
+                unstash: ['foo-unstash']
```

Comment:
`unstash` isn't a string list, is it? Even if the other fields are not required in the tests, wouldn't this be correct?
```suggestion
                unstash: [name: 'foo-unstash']
```

Reply:
No, for unstash we have always only the stash names in the List (... and not a Map with some properties). But for stash we have a Maps in the List.